packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -1,37 +1,47 @@
-/**
- * External dependencies
- */
-import type { ForwardedRef } from 'react';
-import { css } from '@emotion/react';
-
 /**
  * WordPress dependencies
  */
-import { useMemo, useState, useCallback } from '@wordpress/element';
+import {
+	useMemo,
+	useState,
+	useCallback,
+	Children,
+	isValidElement,
+} from '@wordpress/element';
+import deprecated from '@wordpress/deprecated';
 
 /**
  * Internal dependencies
  */
-import {
-	contextConnect,
-	useContextSystem,
-	WordPressComponentProps,
-} from '../../ui/context';
-import { useCx } from '../../utils/hooks/use-cx';
-import { View } from '../../view';
+import { useContextSystem, WordPressComponentProps } from '../../ui/context';
+import { contextConnectWithoutRef } from '../../ui/context/context-connect';
 import { NavigatorContext } from '../context';
 import type {
 	NavigatorProviderProps,
 	NavigatorLocation,
 	NavigatorContext as NavigatorContextType,
 } from '../types';
+import { NavigatorScreen } from '../navigator-screen';
+import { NavigatorContainer } from '../navigator-container';
 
 function UnconnectedNavigatorProvider(
-	props: WordPressComponentProps< NavigatorProviderProps, 'div' >,
-	forwardedRef: ForwardedRef< any >
+	props: WordPressComponentProps< NavigatorProviderProps, 'div', false >
 ) {
-	const { initialPath, children, className, ...otherProps } =
-		useContextSystem( props, 'NavigatorProvider' );
+	const contextProps = useContextSystem( props, 'NavigatorProvider' );
+	const { initialPath } = contextProps;
+	let { children } = contextProps;
+
+	const hasScreenChildren = Children.toArray( children ).some(
+		( child ) => isValidElement( child ) && child.type === NavigatorScreen
```

Comment:
Hm, actually, this doesn't really work in practice because children are often a component that wraps `NavigatorScreen` e.g. `GlobalStylesNavigatorScreen`.